app/src/test/java/com/android/sample/model/map/LocationViewModelTest.kt

```diff
@@ -47,4 +50,69 @@ class LocationViewModelTest {
     // Then
     coVerify(exactly = 0) { mockRepository.search(any(), any(), any()) }
   }
+
```

Comment:
The tests effectively cover both scenarios when permissions are granted and denied, which is essential for verifying location access logic. The use of coVerify(exactly = 0) to check that getCurrentLocation is not called when permission is denied is a good approach.